### PR TITLE
added onpress option to calendar knob

### PR DESCRIPTION
**Pull request by borasumer**
I have added onPress option to the calendar slide knob to be able to open and close the expandableCalendar when we disable the pan option. TODO was already there, and was a coincidence on my needs as well.

### CODE REVIEW

File: src/expandableCalendar/index.js [hunk 1]
```diff
@@ -448,7 +473,13 @@ class ExpandableCalendar extends Component {
 
   renderKnob() {
     // TODO: turn to TouchableOpacity with onPress that closes it
-    return (
+    return this.props.disablePan ? (
+      <TouchableOpacity style={this.style.knobContainer} onPress={() => this.onKnobPress()}>
+        <View style={this.style.knobContainer} pointerEvents={'none'}>
+          <View style={this.style.knob} testID={CALENDAR_KNOB}/>
+        </View>
+      </TouchableOpacity>
```
Comment (khalidkaram) [inline src/expandableCalendar/index.js:481]: When pressing the knob a subtle effect happens:
Untitled
Moving `TouchableOpacity` inside `View` fixed it for me.

Also, I think we can make use of `leftArrowImageSource` and `rightArrowImageSource` with a `90deg` rotation to make it obvious that it's pressable.

```suggestion
      <View style={this.style.knobContainer}>
        <TouchableOpacity
          activeOpacity={0.75}
          style={this.style.knobContainer}
          onPress={() => this.onKnobPress()}>
          <Image
            source={
              isOpen
                ? this.props.leftArrowImageSource
                : this.props.rightArrowImageSource
            }
            style={[this.style.arrowImage, {transform: [{ rotate: '90deg' }]}]}
            testID={CALENDAR_KNOB}
          />
        </TouchableOpacity>
      </View>
```

File: src/expandableCalendar/index.js [hunk 2]
```diff
@@ -474,11 +474,22 @@ class ExpandableCalendar extends Component {
...
+        <TouchableOpacity
+          activeOpacity={0.75}
+          style={this.style.knobContainer}
+          onPress={() => this.onKnobPress()}>
+          <Image
+            source={
+              isOpen
+                ? this.props.leftArrowImageSource
+                : this.props.rightArrowImageSource
+            }
```
Comment (borasumer) [inline src/expandableCalendar/index.js:487]: isOpen is not defined in this closure, are you sure you gave this a try before you submit ?

Comment (khalidkaram) [reply src/expandableCalendar/index.js:487]: oh sorry, there's a missing part before the return :

```
const {position} = this.state;
const isOpen = position === POSITIONS.OPEN;
```

File: src/expandableCalendar/index.js [hunk 3]
```diff
@@ -448,7 +473,24 @@ class ExpandableCalendar extends Component {
 
   renderKnob() {
     // TODO: turn to TouchableOpacity with onPress that closes it
```
Comment (khalidkaram) [inline src/expandableCalendar/index.js:475]:
```suggestion
    // TODO: turn to TouchableOpacity with onPress that closes it
    const {position} = this.state;
    const isOpen = position === POSITIONS.OPEN;
```

File: src/expandableCalendar/index.js [hunk 4]
```diff
@@ -448,7 +473,26 @@ class ExpandableCalendar extends Component {
...
+          activeOpacity={0.75}
+          style={this.style.knobContainer}
+          onPress={() => this.onKnobPress()}>
+          <Image
+            source={
+              isOpen
+                ? this.props.leftArrowImageSource
+                : this.props.rightArrowImageSource
+            }
+            style={[this.style.arrowImage, {transform: [{ rotate: '90deg' }]}]}
```
Comment (khalidkaram) [inline src/expandableCalendar/index.js:490]:
```suggestion
            style={[this.style.arrowImage, {transform: [{rotate: '90deg'}]}]}
```

**Pull request closed without merging**